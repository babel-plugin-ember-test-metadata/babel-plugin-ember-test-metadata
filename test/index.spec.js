const pluginTester = require('babel-plugin-tester').default;
const logger = require('../src/index');
const path = require('path');

pluginTester({
  plugin: logger,
  pluginName: 'logger',
  snapshot: true,
  tests: [
    {
      title: 'changes this code',
      // input to the plugin
      fixture: path.join(__dirname, '__fixtures__/logger', 'code.js'),
      // expected output
      // outputFixture: path.join(__dirname, '__fixtures__/logger', 'output.js'),
    },
  ]
})
