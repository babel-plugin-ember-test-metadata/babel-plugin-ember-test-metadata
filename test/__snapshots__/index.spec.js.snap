// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`logger changes this code: changes this code 1`] = `

function ggg () {

  for (var i = 0; i < 100; i++) {
    console.log(i);
  }

  return 42;
}

function xyz () {

  for (var i = 0; i < 100; i++) {
    console.log(i);
  }

  if (true) {
    console.log('100');
  }

  otherFunction();

}

function outer () {
  function inner () {
    console.log('does something');
  }

  return inner;
}

      ↓ ↓ ↓ ↓ ↓ ↓

function ggg() {
  console.time('ggg');

  for (var i = 0; i < 100; i++) {
    console.log(i);
  }

  console.timeEnd('ggg');
  return 42;
}

function xyz() {
  console.time('xyz');

  for (var i = 0; i < 100; i++) {
    console.log(i);
  }

  if (true) {
    console.log('100');
  }

  otherFunction();
  console.timeEnd('xyz');
}

function outer() {
  console.time('outer');

  function inner() {
    console.time('inner');
    console.log('does something');
    console.timeEnd('inner');
  }

  console.timeEnd('outer');
  return inner;
}


`;
